pkg/planner/cardinality/ndv.go

```diff
@@ -50,17 +50,20 @@ func getTotalRowCount(statsTbl *statistics.Table, colHist *statistics.Column) in
 		return int64(colHist.TotalRowCount())
 	}
 	// If colHist is not fully loaded, we may still get its total row count from other index/column stats.
-	for _, idx := range statsTbl.Indices {
+	totCount := int64(0)
+	statsTbl.ForEachIndex(func(_ int64, idx *statistics.Index) bool {
 		if idx.IsFullLoad() && idx.LastUpdateVersion == colHist.LastUpdateVersion {
-			return int64(idx.TotalRowCount())
```

Comment:
It seems that this changes the original logic, in the old implementation, if we find one, we return immediately.

Reply:
Should we stop here after we find it?